include/oneapi/dpl/experimental/kt/esimd_radix_sort_onesweep.h

```diff
@@ -611,20 +591,19 @@ onesweep(_ExecutionPolicy&& __exec, _Range&& __rng, ::std::size_t __n)
     //types are messy. now all are uint32_t
     const uint32_t SYNC_BUFFER_SIZE = sweep_tg_count * BINCOUNT * STAGES * sizeof(global_hist_t); //bytes
     constexpr uint32_t GLOBAL_OFFSET_SIZE = BINCOUNT * STAGES * sizeof(global_hist_t);
-    const uint32_t LOOKUP_BUFFER_SIZE = sweep_tg_count * BINCOUNT * sizeof(global_hist_t);
-    constexpr uint32_t JOB_QUEUE_SIZE = 256;
-    size_t temp_buffer_size = GLOBAL_OFFSET_SIZE + // global offset
-                              SYNC_BUFFER_SIZE +   // sync buffer
-                              JOB_QUEUE_SIZE +     // dynamic job queue
-                              LOOKUP_BUFFER_SIZE;  // L1 lookup buffer
-
-    uint8_t *tmp_buffer = sycl::malloc_device<uint8_t>(temp_buffer_size, __exec.queue());
-    auto p_global_offset = reinterpret_cast<uint32_t*>(tmp_buffer);
-    auto p_sync_buffer = reinterpret_cast<uint32_t*>(tmp_buffer + GLOBAL_OFFSET_SIZE);
-    auto p_job_queue = reinterpret_cast<uint32_t*>(tmp_buffer + GLOBAL_OFFSET_SIZE + SYNC_BUFFER_SIZE);
-    auto p_lookup =  reinterpret_cast<uint32_t*>(tmp_buffer + GLOBAL_OFFSET_SIZE + SYNC_BUFFER_SIZE + JOB_QUEUE_SIZE);
-    auto p_output = sycl::malloc_device<uint32_t>(__n, __exec.queue());
-    auto e_init = __exec.queue().memset(tmp_buffer, 0, temp_buffer_size);
+    size_t temp_buffer_size = GLOBAL_OFFSET_SIZE + SYNC_BUFFER_SIZE;
+
+    ::std::shared_ptr<uint8_t> tmp_buffer(sycl::malloc_device<uint8_t>(temp_buffer_size, __exec.queue()),
+                                          [__exec](uint8_t* ptr) { sycl::free(ptr, __exec.queue()); });
+    auto p_tmp_buffer = tmp_buffer.get();
+    auto e_init = __exec.queue().memset(p_tmp_buffer, 0, temp_buffer_size);
+
+    auto p_global_offset = reinterpret_cast<uint32_t*>(p_tmp_buffer);
+    auto p_sync_buffer = reinterpret_cast<uint32_t*>(p_tmp_buffer + GLOBAL_OFFSET_SIZE);
+
+    ::std::shared_ptr<uint32_t> p_output(sycl::malloc_device<uint32_t>(__n, __exec.queue()),
```

Comment:
I think it is better to leave raw pointers and free calls, because:
- There is no difficult memory management; we create and destroy these object in the same function
- It does not really improve readability
- It incurs overhead